### PR TITLE
Registration page with pepcus core

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/components/common/fields/SelectList.jsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+  const title = get(schema, 'title', name);
+
+  const defaultValue = get(schema, 'default', '');
+
+  const enumNames = get(schema, 'enumNames', []);
+
+  const enums = get(schema, 'enum', []);
+
+  const titleStyle = get(uiSchema, 'ui:options.style.titleStyle');
```
Comment (Nikmahajan) [inline src/components/common/fields/SelectList.jsx]: Extract a single node at a time to get nest objects.

File: src/components/common/fields/InputField.jsx [hunk 2]
```diff
@@ -55,34 +56,35 @@ function InputField(props) {
 
   function handleOnChange(event) {
     // form validation for required filed not work is case the value of field is empty or null
-    const value = event ? event.target.value : '';
+    const values = event ? event.target.value : '';
```
Comment (Nikmahajan) [inline src/components/common/fields/InputField.jsx]: Why it is renamed as plural?

File: src/components/common/fields/SelectList.jsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+          </SelectOptionStyled>),
+      );
+    }
+    return enums.map((iterator, index) =>
+      (
+        <SelectOptionStyled value={iterator} key={index}>
+          {iterator}
+        </SelectOptionStyled>),
+    );
+  };
```
Comment (Nikmahajan) [inline src/components/common/fields/SelectList.jsx]: Can be simplified:- 

```suggestion
  return enums.map((iterator, index) =>
        (
          <SelectOptionStyled value={iterator} key={index}>
            {enumNames[index] || iterator}
          </SelectOptionStyled>),
      );
  
      };
```

File: src/components/common/fields/SelectList.jsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        <Col {...colProps}>
+          <Row width="auto" margin="0 0 0 0">
+            <label style={{ ...defaultTitleStyle, ...titleStyle }}>
+              {title}{required ? '*' : ''}
+            </label>
+          </Row>
+          <Row width="100%" margin="0 0 0 0">
+            <SelectStyled
+              disabled={disabled}
+              style={{ ...fieldStyle }}
```
Comment (Nikmahajan) [inline src/components/common/fields/SelectList.jsx]: Unnecessary spread operators.

File: src/components/common/fields/SelectList.jsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+              minWidth="100%"
+              id={idSchema.$id}
+              onChange={handleOnChange}
+              required={required}
+              defaultValue={defaultValue}
+            >
+              <SelectOptionStyled
+                value=""
+                disabled="disabled"
+                style={{ display: 'none', color: 'red' }}
```
Comment (Nikmahajan) [inline src/components/common/fields/SelectList.jsx]: This component is already been styled above, Makes no sense of passing static style object.

File: src/components/common/fields/SelectList.jsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+              width="100%"
+              minWidth="100%"
+              id={idSchema.$id}
+              onChange={handleOnChange}
+              required={required}
+              defaultValue={defaultValue}
+            >
+              <SelectOptionStyled
+                value=""
+                disabled="disabled"
```
Comment (Nikmahajan) [inline src/components/common/fields/SelectList.jsx]: The `disabled `prop should be a boolean value.

File: src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+import fields from 'components/common/fields';
+import RedirectToRoute from './RedirectToRoute';
+import SuccessMessagePopup from './SuccessMessagePopup';
+
+const BoxStyled = styled(Box)`
+ align-items: center;
+ overflow-x: hidden;
+ overflow-y: auto;
+ background-color: ${getThemeProps('palette.policyMuted.color')};
+ ${({ theme }) => theme.media.down('md')`
```
Comment (Nikmahajan) [inline src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx]: Empty media styling

File: src/components/common/fields/SelectList.jsx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+  const colProps = get(props, 'content.props.uiSchema.ui:options.col', {});
+
+  const placeholder = get(uiSchema, 'ui:placeholder', '');
+
+  function handleOnChange(event) {
+    // form validation for required filed not work is case the value of field is empty or null
+    const values = event ? event.target.value : '';
+
+    if (isEmpty(values) || values === null) {
+      onChange(undefined);
```
Comment (pepcus-pratik) [inline src/components/common/fields/SelectList.jsx]:
```suggestion
      onChange();
```

File: src/components/common/fields/SelectList.jsx [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+
+  const colProps = get(props, 'content.props.uiSchema.ui:options.col', {});
+
+  const placeholder = get(uiSchema, 'ui:placeholder', '');
+
+  function handleOnChange(event) {
+    // form validation for required filed not work is case the value of field is empty or null
+    const values = event ? event.target.value : '';
+
+    if (isEmpty(values) || values === null) {
```
Comment (pepcus-pratik) [inline src/components/common/fields/SelectList.jsx]:
```suggestion
    if (isEmpty(values)) {
```

File: src/components/common/fields/SelectList.jsx [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+      onChange(undefined);
+    } else {
+      onChange(values);
+    }
+  }
+
+  const renderOptionStatements = () => {
```
Comment (pepcus-pratik) [inline src/components/common/fields/SelectList.jsx]:
```suggestion
  const renderOptions = () => {
```

File: src/components/common/fields/SelectList.jsx [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    if (enumNames) {
+      return enums.map((iterator, index) =>
+        (
+          <SelectOptionStyled value={iterator} key={index}>
+            {enumNames[index]}
+          </SelectOptionStyled>),
+      );
+    }
+    return enums.map((iterator, index) =>
```
Comment (pepcus-pratik) [inline src/components/common/fields/SelectList.jsx]: Rename `iterator`

File: src/components/routeComponents/index.js [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import SplashPage from './splashPage';
-import StudentRegistrationForm from './StudentRegistrationForm';
+import MemberRegistrationForm from './memberRegistraionForm';
```
Comment (pepcus-pratik) [inline src/components/routeComponents/index.js]:
```suggestion
import MemberRegistrationForm from './memberRegistrationForm';
```

File: src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+   * @param {Object} event
+   */
+  onChange = (event) => {
+    const { member } = this.state;
+    this.setState({
+      member: {
+        ...member,
+        ...event.formData,
+      },
+      notHasAnError: isEmpty(event.errors),
```
Comment (pepcus-pratik) [inline src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx]:
```suggestion
      hasError: isEmpty(event.errors),
```

File: src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+  /**
+   * Method check the validation for individual form fields
+   * @param {Object} formData
+   * @param {Object} errors
+   * @return {Object} errors
+   */
+  validate = (formData, errors) => {
+    const { validation } = this.state.formConfig;
+
+    return verifyFormDataValidations({ formData, errors, validate: validation });
```
Comment (pepcus-pratik) [inline src/components/routeComponents/memberRegistraionForm/MemberRegistrationForm.jsx]:
```suggestion
    return validateForm({ formData, errors, validate: validation });
```

**Pull request merged**